runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkExecutableStageFunction.java

```diff
@@ -141,8 +144,17 @@ public void mapPartition(
 
   @Override
   public void close() throws Exception {
-    try (AutoCloseable bundleFactoryCloser = stageBundleFactory) {}
-    try (AutoCloseable closable = stageContext) {}
+    // close may be called multiple times when an exception is thrown
```

Comment:
Should we synchronize close then? The calls could be interleaved.

Reply:
It's not called concurrently, it is just being called again when the first call throws an exception.